iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -207,7 +203,81 @@ struct FftOpConversion : public OpConversionPattern<mhlo::FftOp> {
     return success();
   }
 };
-}  // namespace
+
+// We need to convert func ops in order to convert types.
+class BuiltinFuncOpPattern : public OpConversionPattern<FuncOp> {
+  using OpConversionPattern<FuncOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      FuncOp srcOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    FunctionType srcFuncType = srcOp.getType();
+    TypeConverter::SignatureConversion signatureConversion(
+        srcOp.getNumArguments());
+
+    // Convert function arguments.
+    for (unsigned i = 0, e = srcFuncType.getNumInputs(); i < e; ++i) {
+      if (failed(getTypeConverter()->convertSignatureArg(
+              i, srcFuncType.getInput(i), signatureConversion))) {
+        return rewriter.notifyMatchFailure(srcOp, "argument failed to convert");
+      }
+    }
+
+    // Convert function results.
+    SmallVector<Type, 1> convertedResultTypes;
```

Comment:
Is an explicit N here actually useful/required? 1 is probably the most common number of return types, but not sure it's worth it.

> In the absence of a well-motivated choice for the number of inlined elements N, it is recommended to use SmallVector<T> (that is, omitting the N). This will choose a default number of inlined elements reasonable for allocation on the stack (for example, trying to keep sizeof(SmallVector<T>) around 64 bytes).

https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h